gulpfile.js

```diff
@@ -88,34 +88,35 @@ var paths = {
   },
 
   vendorChallenges: [
-    'public/bower_components/jshint/dist/jshint.js',
-    'public/bower_components/chai/chai.js',
-    'public/bower_components/CodeMirror/lib/codemirror.js',
-    'public/bower_components/CodeMirror/addon/comment/comment.js',
-    'public/bower_components/CodeMirror/addon/edit/closebrackets.js',
-    'public/bower_components/CodeMirror/addon/edit/matchbrackets.js',
-    'public/bower_components/CodeMirror/addon/lint/lint.js',
-    'public/bower_components/CodeMirror/addon/lint/javascript-lint.js',
-    'public/bower_components/CodeMirror/mode/javascript/javascript.js',
-    'public/bower_components/CodeMirror/mode/xml/xml.js',
-    'public/bower_components/CodeMirror/mode/css/css.js',
-    'public/bower_components/CodeMirror/mode/htmlmixed/htmlmixed.js',
+    'node_modules/jshint/dist/jshint.js',
+    'node_modules/chai/chai.js',
+    'node_modules/codemirror/lib/codemirror.js',
+    'node_modules/codemirror/addon/comment/comment.js',
+    'node_modules/codemirror/addon/edit/closebrackets.js',
+    'node_modules/codemirror/addon/edit/matchbrackets.js',
+    'node_modules/codemirror/addon/lint/lint.js',
+    'node_modules/codemirror/addon/lint/javascript-lint.js',
+    'node_modules/codemirror/mode/javascript/javascript.js',
+    'node_modules/codemirror/mode/xml/xml.js',
+    'node_modules/codemirror/mode/css/css.js',
+    'node_modules/codemirror/mode/htmlmixed/htmlmixed.js',
```

Comment:
Yup, that seems right. However:

`require.resolve` relies on [lookup order used by `require`](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders). If you are calling `require.resolve` in gulpfile, the first directory it looks in will be the global `node_modules` for the project.

For example, 
`require.resolve('jquery)` in gulpfile will return path for the jquery version that lives in `FreeCodeCamp/node_modules/jquery/`

So these two lines point to the same file when we write it in gulpfile:
`'node_modules/jquery/dist/jquery.js'`
and
`require.resolve('jquery')`

To your earlier point, npm3 is non-deterministic, and `FreeCodeCamp/node_modules/jquery/` may potentially contain the wrong version of `jquery`. That's basically why I'm advocating for  using shrinkwrap with npm. Shrinkwrap locks the dependencies of dependencies, so `FreeCodeCamp/node_modules/jquery/` will always contain the same jquery version regardless on the environment.